theories/structure/FinType.ec

```diff
@@ -62,3 +62,9 @@ case=> x y; rewrite count_uniq_mem.
 + by apply/b2i_eq1/allpairsP; exists (x, y); rewrite !(FT1.enumP, FT2.enumP).
 qed.
 end FinProdType.
+
+clone FinType as UnitFinType with
+  type t = unit,
```

Comment:
If I understand correctly, then the difference between `<=` and `=` is whether the definition is unfolded in statements (e.g. `enum_spec`). So IMHO it makes sense to use `<=` for `t` but not for `enum`. After all, `enum` should be treated abstractly in proofs about finite types. Did I miss something?

Reply:
`<=` substitutes the RHS for the left-hand symbol in definitions and lemmas, but also keeps the symbol definition (this is useful if you later want to merge the cloned and instantiated theory with another instance).

The point you make about wanting to keep `enum` abstract is interesting, but 1) we're now no longer talking about finite types, we're talking about the `unit` type—the proofs about finite types are indeed already done with an abstract `enum` symbol; and 2) the discussion is kind of moot unless we also make `enum` `opaque` when we instantiate it, and have mechanisms to break opacity on demand.

Depending on whether I misinterpreted your intent, and my point 1) is completely off topic, I don't know if support for 2) would be useful. (You can of course always define an `opaque` symbol that wraps `[tt]` and use it to instantiate, but that's silly.)

Reply:
Well, for `unit` printing `[tt]` is fine, my point was merely that, in general, printing the type (e.g. `unit` instead of `UnitFinType.t`) can be expected to make things easier to read. This is less clear for `enum`. I'd prefer a statement involving `SomeLargeFinType.enum` rather than an obscure (possibly large) list. I wasn't talking about real abstraction barriers. Anyway, I hope that much of this becomes moot with the introduction of type classes.

Reply:
That's a solid point. I guess we're good to merge.